### DISCUSSION TITLE
Checkout function bug with 3-page checkout prevents blocking

### DISCUSSION THREAD

**Discussion by MerchantlyApps** [Shopify/function-examples#503] [Feedback: Cart and Checkout Validation API]
Hi,

We’ve noticed a bug with our clients using 3-page checkout, where the checkout validation function is not catching the error on Line1 .. 100% of the time. In some instances a customer can click to Step2, but when they click back to Step1 - the error shows fine.

Here is a reproduction video of the issue: 

https://monosnap.com/direct/bCRMQOjxgvIQcuBrRd2CTTGSDxC9CM

For some reason, Shopify doesn't immediately show our error and allows the user to move to the next checkout step.

When we go back to the previous step, the error is immediately shown, and now Shopify does not allow us to move forward to the next checkout step.

After changing the address, Shopify allows us to move forward again, although the PO Box error is still present.

==
Expected Result:

If the entered text matches the function rule, it should block checkout 100% of the time and prevent the user from clicking to the next step. Currently this seems to only work sometimes and breaks often with 3-page checkout.

**Comment by bithaolee** [comment]
Had the same problem.
